tests/unit/s2n_mem_allocator_test.c

```diff
@@ -185,6 +185,10 @@ int main(int argc, char **argv)
             /* This is the client process, close the server end of the pipe */
             EXPECT_SUCCESS(s2n_io_pair_close_one_end(&io_pair, S2N_SERVER));
 
+            free(cert_chain_pem);
+            free(private_key_pem);
```

Comment:
I am also very curious to know if DEFER_CLEANUP works instead of just freeing this stuff after forking.

Reply:
I think using DEFER_CLEANUP would work here, but the mock_client() would have to change to call `return 0` in place of `exit()`. Because I suspect the child needs to execute the cleanup callback, which it wouldn't right now, since it just exits the process. So interesting!